src/xbot/common/math/FieldPose.java

```diff
@@ -0,0 +1,127 @@
+package xbot.common.math;
+
+public class FieldPose {
+
+    private double m;
+    private double b;
+    private ContiguousHeading heading;
+    private final XYPair fieldPosition;
+    
+    public FieldPose(XYPair point, ContiguousHeading heading) {
+        m = degreesToSlope(heading.getValue());
+        this.fieldPosition = point;
+        this.heading = heading;
+        
+        // if we already have the y-intercept, avoid division by 0.
+        if ((point.x == 0) || (Math.abs(m) < 0.01)) {
+            b = point.y;
+        } else {
+            b = point.y / (point.x * m); 
+        }   
+    }
+    
+    public FieldPose(XYPair point, double slope, boolean positiveAngle) {
+        this.fieldPosition = point;
+        m = slope;
+        b = point.y - (point.x * m);
+        
+        // if slope negative, but positive angle, we are in quadrant II
+        if (slope < 0 && positiveAngle) {
+            heading = new ContiguousHeading(slopeToDegrees(m) + 180);
+        }
+        // if slope positive, but negative angle, we are in quadrant III
+        else if (slope > 0 && !positiveAngle) {
+            heading = new ContiguousHeading(slopeToDegrees(m) - 180);
+        } else {
+            heading = new ContiguousHeading(slopeToDegrees(m));
+        }
+       
+    }
+    
+    public double getSlope() {
+        return m;
+    }
+    
+    public ContiguousHeading getHeading() {
+        return heading;
+    }
+    
+    public double getIntercept() {
+        return b;
+    }
+    
+    public XYPair getPoint() {
+        return fieldPosition;
+    }
+    
+    private double degreesToSlope(double degrees) {
+        double deltaY = Math.sin(Math.toRadians(degrees));
+        double deltaX = Math.cos(Math.toRadians(degrees));
+        
+        return deltaY/deltaX;
+    }
+    
+    private double slopeToDegrees(double slope) {
+        double x = 1;
+        if (Math.abs(this.m) > 90) {
+            x = -1;
+        }
+        
+        double rads = Math.atan2(this.m, x);
+        
+        return Math.toDegrees(rads);
+    }
+    
+    public FieldPose getLinePerpendicularToPoint(XYPair point) {
+        // for lines of zero slope, this is not great
+        
+        double perp_m = 0;
+        
+        if (m == 0) {
+            // if we have no slope, instead of a line with infinite slope, we just use a very large slope, 
+            // like a million.
+            perp_m = 1000000;
+        } else {
+            perp_m = -1/m; 
+        }
+        
+        return new FieldPose(point, perp_m,  heading.getValue() > 0);
+    }
+    
+    public double getY(double x) {
+        return x*m+b;
+    }
+    
+    private XYPair getIntersectionWithLine(FieldPose line) {
+        // calculate X point where they meet
+        
+        double x_intersect = (line.b - this.b) / (this.m - line.m);
```

Comment:
When two lines meet, it's appropriate to call that an "intersection"
http://math.stackexchange.com/questions/216151/difference-between-intercept-and-intersect